File: governance/config/hash_exclusion_config.py
from typing import Dict, List

# --- Configuration for Canonicalization Exclusion ---

# Defines paths within structured payloads that should be *excluded* from 
# canonicalization and cryptographic hashing.
# Exclusion is necessary for transient fields added between processing stages 
# (e.g., timestamps, stage IDs, runtime metrics).
# Paths must be defined using dot notation (e.g., 'metadata.timestamp').

HASH_EXCLUSION_CONFIG: Dict[str, List[str]] = {
    # M-02: Manifest Payload Type (primary governance artifact)
    "M-02": [
        "system_metadata.timestamp_staged",
        "system_metadata.stage3_reviewer_id",
        "transient_runtime_id"
    ],
    # P-01: Processing Report Payload Type
    "P-01": [
        "analysis_runtime.cpu_utilization",
        "analysis_runtime.run_duration_ms",
        "analysis_runtime.execution_node_id"
    ]
}

File: governance/config/hashExclusionConfig.js
/**
 * Configuration map defining fields that must be excluded from canonicalization 
 * before cryptographic hashing, based on G0_Rules (e.g., transient timestamps, signatures).
 * Keys are payload types (e.g., 'TRANSACTION'), values are arrays of dot-notation paths.
 */
const HASH_EXCLUSION_CONFIG = {
    // Placeholder configuration for execution stability
    'TRANSACTION': [
        'metadata.timestamp',
        'signature'
    ],
    'STATE_UPDATE': [
        'auditTrail.lastModified'
    ]
};

module.exports = HASH_EXCLUSION_CONFIG;